diff -u includes/DefaultSettings.php includes/DefaultSettings.php
--- includes/DefaultSettings.php	2012-08-31 00:25:34.000000000 +0200
+++ includes/DefaultSettings.php	2012-09-02 12:47:19.890181785 +0200
@@ -4311,7 +4311,7 @@
  * (results are passed around in OpenSearch format)
  * Requires $wgEnableOpenSearchSuggest = true;
  */
-$wgEnableMWSuggest = false;
+$wgEnableMWSuggest = true;
 
 /**
  * Enable OpenSearch suggestions requested by MediaWiki. Set this to
