diff -rup a/include/tests_boot_services b/include/tests_boot_services
--- a/include/tests_boot_services       2014-03-07 15:58:54.000000000 +0100
+++ b/include/tests_boot_services       2014-03-20 11:27:38.544783376 +0100
@@ -39,10 +39,10 @@
             Display --indent 4 --text "- Checking presence GRUB... " --result "OK" --color GREEN
             if [ -f /boot/grub/grub.conf ]; then GRUBCONFFILE="/boot/grub/grub.conf"; else GRUBCONFFILE="/boot/grub/menu.lst"; fi
             logtext "Found file ${GRUBCONFFILE}, proceeding with tests."
-            FIND=`cat ${GRUBCONFFILE} | grep 'password --md5' | grep -v '^#'`
+            FIND=`cat ${GRUBCONFFILE} | egrep 'password --[md5|encrypted]' | grep -v '^#'`
             if [ "${FIND}" = "" ]; then
                 Display --indent 6 --text "- Checking for password protection..." --result WARNING --color RED
-                logtext "Result: Didn't find MD5 hashed password line in GRUB boot file!"
+                logtext "Result: Didn't find MD5 or SHA hashed password line in GRUB boot file!"
                 logtext "Risk: user can switch to single user mode by editing current menu items or bypassing them."
                 logtext "Additional information: Do NOT use a plaintext password, since the grub.conf or menu.lst file is most likely to be world readable!"
                 logtext "If an unsecured OS like DOS is used, add 'lock' below that entry and setup a password with the password option, to prevent direct system access."
