From 926b133a5207ea9113f5b2ca9e4c440f8bad59ec Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 10 Mar 2010 12:08:35 +0300
Subject: [PATCH 2/6] Remove assignment of (not parsed) to jitter

---
 plugins-scripts/check_ntp.pl |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/plugins-scripts/check_ntp.pl b/plugins-scripts/check_ntp.pl
index 5c87e0a..0733a63 100755
--- a/plugins-scripts/check_ntp.pl
+++ b/plugins-scripts/check_ntp.pl
@@ -313,7 +313,6 @@ if ($have_ntpq) {
 				}
 			} else {
 				print "No match!\n" if $verbose;
-				$jitter = '(not parsed)';
 			}
 			
 		}
-- 
1.7.2.3

