diff -rup a/active-response/firewall-drop.sh b/active-response/firewall-drop.sh
--- a/active-response/firewall-drop.sh	2013-10-29 19:13:44.000000000 +0100
+++ b/active-response/firewall-drop.sh	2014-02-02 11:52:33.721638365 +0100
@@ -32,7 +32,8 @@ IP=$3
 PWD=`pwd`
 LOCK="${PWD}/fw-drop"
 LOCK_PID="${PWD}/fw-drop/pid"
-
+TIMESTAMP=`date +%s`
+OHOME="/var/ossec/"
 
 LOCAL=`dirname $0`;
 cd $LOCAL
@@ -143,9 +144,15 @@ if [ "X${UNAME}" = "XLinux" ]; then
    if [ "x${ACTION}" = "xadd" ]; then
       ARG1="-I INPUT -s ${IP} -j DROP"
       ARG2="-I FORWARD -s ${IP} -j DROP"
+      # ASL (track IP for web gui)
+      touch $OHOME/var/shun-${TIMESTAMP}-${IP}-$5
    else
       ARG1="-D INPUT -s ${IP} -j DROP"
       ARG2="-D FORWARD -s ${IP} -j DROP"
+      # ASL (track IP for web gui)
+      if [ -f $OHOME/var/*${IP}* ]; then
+        rm -f  $OHOME/var/*${IP}*
+      fi
    fi
    
    # Checking if iptables is present
