diff -ru ossec-hids-2.6/src/os_dbd/alert.c ossec-hids-2.6-new/src/os_dbd/alert.c
--- ossec-hids-2.6/src/os_dbd/alert.c	2011-07-11 15:36:59.000000000 -0400
+++ ossec-hids-2.6-new/src/os_dbd/alert.c	2011-08-23 10:22:11.000000000 -0400
@@ -187,7 +187,15 @@
     i = 0;
     while(al_data->log[i])
     {
-        fulllog = os_LoadString(fulllog, al_data->log[i]);
+        char templog[strlen(al_data->log[i])+2];
+        if (al_data->log[i+1]) {
+            sprintf(templog, "%s\n", al_data->log[i]);
+        }
+        else {
+            sprintf(templog, "%s", al_data->log[i]);
+        }
+        fulllog = os_LoadString(fulllog, templog);
+//      fulllog = os_LoadString(fulllog, al_data->log[i]);
         i++;
     }
     osdb_escapestr(fulllog);
diff -ru ossec-hids-2.6/src/os_dbd/db_op.c ossec-hids-2.6-new/src/os_dbd/db_op.c
--- ossec-hids-2.6/src/os_dbd/db_op.c	2011-07-11 15:36:59.000000000 -0400
+++ ossec-hids-2.6-new/src/os_dbd/db_op.c	2011-08-23 10:22:11.000000000 -0400
@@ -46,7 +46,7 @@
     {
         return;
     }
-    
+
     while(*str)
     {
         if(*str == '\'')
diff -ru ossec-hids-2.6/src/os_dbd/db_op.h ossec-hids-2.6-new/src/os_dbd/db_op.h
--- ossec-hids-2.6/src/os_dbd/db_op.h	2011-07-11 15:36:59.000000000 -0400
+++ ossec-hids-2.6-new/src/os_dbd/db_op.h	2011-08-23 10:22:11.000000000 -0400
@@ -53,7 +53,8 @@
 static const unsigned char insert_map[] =
 {
     '\000', '\000', '\002', '\003', '\004', '\005', '\006', '\007',
-    '\010', '\011', '\012', '\013', '\014', '\015', '\016', '\017',
+//  Modified to allow new lines - \012
+    '\010', '\011', '\001', '\013', '\014', '\015', '\016', '\017',
     '\020', '\021', '\022', '\023', '\024', '\025', '\026', '\027',
     '\030', '\031', '\032', '\033', '\034', '\035', '\036', '\037',
     '\001', '\001', '\001', '\001', '\001', '\001', '\001', '\047',
