diff -ru ossec-hids-2.6-orig/src/os_dbd/mysql.schema ossec-hids-2.6/src/os_dbd/mysql.schema
--- ossec-hids-2.6-orig/src/os_dbd/mysql.schema	2011-07-11 15:36:59.000000000 -0400
+++ ossec-hids-2.6/src/os_dbd/mysql.schema	2012-04-16 17:19:33.000000000 -0400
@@ -16,8 +16,7 @@
     (
     cat_id      SMALLINT    UNSIGNED NOT NULL   AUTO_INCREMENT,
     cat_name    VARCHAR(32) NOT NULL            UNIQUE,
-    PRIMARY KEY (cat_id),
-    INDEX       (cat_name)
+    PRIMARY KEY (cat_id)
     );
 
 CREATE TABLE signature 
