diff -up ossec-hids-2.4/etc/ossec-agent.conf.orig ossec-hids-2.4/etc/ossec-agent.conf
--- ossec-hids-2.4/etc/ossec-agent.conf.orig	2006-05-06 15:40:18.000000000 +0200
+++ ossec-hids-2.4/etc/ossec-agent.conf	2010-04-16 11:09:39.000000000 +0200
@@ -6,25 +6,40 @@
   </client>
 
   <syscheck>
+    <!-- <disabled>yes</disabled> -->
+
     <!-- Frequency that syscheck is executed -- default every 2 hours -->
     <frequency>7200</frequency>
     
     <!-- Directories to check  (perform all possible verifications) -->
     <directories check_all="yes">/etc,/usr/bin,/usr/sbin</directories>
     <directories check_all="yes">/bin,/sbin</directories>
+    <directories check_all="yes">/usr/lib64,/usr/lib,/lib64,/lib</directories>
+    <directories check_all="yes">/usr/local/bin,/usr/local/sbin</directories>
+    <directories check_all="yes">/usr/local/lib64,/usr/local/lib</directories>
 
     <!-- Files/directories to ignore -->
     <ignore>/etc/mtab</ignore>
+    <ignore>/etc/mnttab</ignore>
     <ignore>/etc/hosts.deny</ignore>
     <ignore>/etc/mail/statistics</ignore>
-    <ignore>/etc/random-seed</ignore>
     <ignore>/etc/adjtime</ignore>
-    <ignore>/etc/httpd/logs</ignore>
+    <ignore>/etc/cups/certs</ignore>
+    <ignore>/etc/dumpdates</ignore>
   </syscheck>
 
   <rootcheck>
+    <!-- <disabled>yes</disabled> -->
+
+    <!-- Frequency that rootcheck is executed - every 12 hours -->
+    <frequency>43200</frequency>
+
+    <!-- Local of the rootkit_files and rootkit_trojans files -->
     <rootkit_files>/var/ossec/etc/shared/rootkit_files.txt</rootkit_files>
     <rootkit_trojans>/var/ossec/etc/shared/rootkit_trojans.txt</rootkit_trojans>
+    <!-- <system_audit>/var/ossec/etc/shared/system_audit_rcl.txt</system_audit> -->
+    <!-- <system_audit>/var/ossec/etc/shared/cis_rhel_linux_rcl.txt</system_audit> -->
+    <!-- <system_audit>/var/ossec/etc/shared/cis_rhel5_linux_rcl.txt</system_audit> -->
   </rootcheck>
 
   <localfile>
@@ -34,31 +49,38 @@
 
   <localfile>
     <log_format>syslog</log_format>
-    <location>/var/log/authlog</location>
+    <location>/var/log/secure</location>
   </localfile>
 
   <localfile>
     <log_format>syslog</log_format>
-    <location>/var/log/secure</location>
+    <location>/var/log/maillog</location>
   </localfile>
 
+  <!--
   <localfile>
-    <log_format>syslog</log_format>
-    <location>/var/log/xferlog</location>
+    <log_format>snort-full</log_format>
+    <location>/var/log/snort/alert</location>
   </localfile>
 
   <localfile>
-    <log_format>syslog</log_format>
-    <location>/var/log/maillog</location>
+    <log_format>snort-fast</log_format>
+    <location>/var/log/snort/alert</location>
   </localfile>
 
   <localfile>
     <log_format>apache</log_format>
-    <location>/var/www/logs/access_log</location>
+    <location>/var/log/httpd/access_log</location>
   </localfile>
 
   <localfile>
     <log_format>apache</log_format>
-    <location>/var/www/logs/error_log</location>
+    <location>/var/log/httpd/error_log</location>
+  </localfile>
+
+  <localfile>
+    <log_format>postgresql_log</log_format>
+    <location>/var/log/postgresql/postgresql.log</location>
   </localfile>
+  -->
 </ossec_config>
diff -up ossec-hids-2.4/etc/ossec-server.conf.orig ossec-hids-2.4/etc/ossec-server.conf
--- ossec-hids-2.4/etc/ossec-server.conf.orig	2010-03-02 18:09:00.000000000 +0100
+++ ossec-hids-2.4/etc/ossec-server.conf	2010-04-16 11:29:33.000000000 +0200
@@ -68,27 +68,41 @@
     <include>local_rules.xml</include>
   </rules>
 
-
   <syscheck>
+    <!-- <disabled>yes</disabled> -->
+
     <!-- Frequency that syscheck is executed -- default every 20 hours -->
     <frequency>72000</frequency>
     
     <!-- Directories to check  (perform all possible verifications) -->
     <directories check_all="yes">/etc,/usr/bin,/usr/sbin</directories>
     <directories check_all="yes">/bin,/sbin</directories>
+    <directories check_all="yes">/usr/lib64,/usr/lib,/lib64,/lib</directories>
+    <directories check_all="yes">/usr/local/bin,/usr/local/sbin</directories>
+    <directories check_all="yes">/usr/local/lib64,/usr/local/lib</directories>
 
     <!-- Files/directories to ignore -->
     <ignore>/etc/mtab</ignore>
+    <ignore>/etc/mnttab</ignore>
     <ignore>/etc/hosts.deny</ignore>
     <ignore>/etc/mail/statistics</ignore>
-    <ignore>/etc/random-seed</ignore>
     <ignore>/etc/adjtime</ignore>
-    <ignore>/etc/httpd/logs</ignore>
+    <ignore>/etc/cups/certs</ignore>
+    <ignore>/etc/dumpdates</ignore>
   </syscheck>
 
   <rootcheck>
+    <!-- <disabled>yes</disabled> -->
+
+    <!-- Frequency that rootcheck is executed - every 12 hours -->
+    <frequency>43200</frequency>
+
+    <!-- Local of the rootkit_files and rootkit_trojans files -->
     <rootkit_files>/var/ossec/etc/shared/rootkit_files.txt</rootkit_files>
     <rootkit_trojans>/var/ossec/etc/shared/rootkit_trojans.txt</rootkit_trojans>
+    <!-- <system_audit>/var/ossec/etc/shared/system_audit_rcl.txt</system_audit> -->
+    <!-- <system_audit>/var/ossec/etc/shared/cis_rhel_linux_rcl.txt</system_audit> -->
+    <!-- <system_audit>/var/ossec/etc/shared/cis_rhel5_linux_rcl.txt</system_audit> -->
   </rootcheck>
 
   <global>
@@ -129,7 +143,6 @@
     <timeout_allowed>yes</timeout_allowed>
   </command>  
 
-
   <!-- Active Response Config -->
   <active-response>
     <!-- This response is going to execute the host-deny
@@ -155,7 +168,6 @@
   </active-response>  
 
   <!-- Files to monitor (localfiles) -->
-
   <localfile>
     <log_format>syslog</log_format>
     <location>/var/log/messages</location>
@@ -163,31 +175,43 @@
 
   <localfile>
     <log_format>syslog</log_format>
-    <location>/var/log/authlog</location>
+    <location>/var/log/secure</location>
   </localfile>
 
   <localfile>
     <log_format>syslog</log_format>
-    <location>/var/log/secure</location>
+    <location>/var/log/maillog</location>
   </localfile>
 
+  <!--
   <localfile>
     <log_format>syslog</log_format>
     <location>/var/log/xferlog</location>
   </localfile>
 
   <localfile>
-    <log_format>syslog</log_format>
-    <location>/var/log/maillog</location>
+    <log_format>snort-full</log_format>
+    <location>/var/log/snort/alert</location>
+  </localfile>
+
+  <localfile>
+    <log_format>snort-fast</log_format>
+    <location>/var/log/snort/alert</location>
   </localfile>
 
   <localfile>
     <log_format>apache</log_format>
-    <location>/var/www/logs/access_log</location>
+    <location>/var/log/httpd/access_log</location>
   </localfile>
 
   <localfile>
     <log_format>apache</log_format>
-    <location>/var/www/logs/error_log</location>
+    <location>/var/log/httpd/error_log</location>
+  </localfile>
+
+  <localfile>
+    <log_format>postgresql_log</log_format>
+    <location>/var/log/postgresql/postgresql.log</location>
   </localfile>
+  -->
 </ossec_config>
