diff -rup a/src/hf/gl_preview.c b/src/hf/gl_preview.c
--- a/src/hf/gl_preview.c	2010-11-28 04:34:44.000000000 +0100
+++ b/src/hf/gl_preview.c	2014-05-28 14:37:00.048364287 +0200
@@ -30,7 +30,7 @@ in Linux Magazine France, n. 33-34.
  /* 2005-11: change library from gtkglarea to gtkglext
  */
 
-
+#include <locale.h>
 #include <GL/gl.h>
 #include <GL/glu.h>
 #include <gtk/gtk.h>
diff -rup a/src/utils/menus_n_tools.c b/src/utils/menus_n_tools.c
--- a/src/utils/menus_n_tools.c	2010-07-31 06:10:44.000000000 +0200
+++ b/src/utils/menus_n_tools.c	2014-05-28 13:50:01.853495106 +0200
@@ -27,7 +27,7 @@
 // #define _(x) pr(x)
 
 gchar *pr(gchar *x) {
-	printf(x);
+	printf("%s", x);
 	return (gchar *) gettext(x);
 }
 
