diff -rup a/setup.py b/setup.py
--- a/setup.py	2014-05-02 06:43:53.000000000 +0200
+++ b/setup.py	2014-09-20 11:03:00.290405876 +0200
@@ -53,7 +53,7 @@ setup(
                 'HTML output!',
     long_description=__doc__,
     scripts = ['src/TermRecord'],
-    data_files = [('/usr/local/share/TermRecord/templates',
+    data_files = [('/usr/share/TermRecord/templates',
                                                  ['templates/dynamic.jinja2',
                                                   'templates/base.jinja2',
                                                   'templates/static.jinja2'])],
diff -rup a/src/TermRecord b/src/TermRecord
--- a/src/TermRecord	2014-05-02 06:43:53.000000000 +0200
+++ b/src/TermRecord	2014-09-20 11:03:25.986302705 +0200
@@ -33,7 +33,7 @@ from jinja2.environment import  Environm
 
 
 
-DEFAULT_TEMPLATE    =   '/usr/local/share/TermRecord/templates/static.jinja2'
+DEFAULT_TEMPLATE    =   '/usr/share/TermRecord/templates/static.jinja2'
 TTYREC              =   'ttyrec'
 
 
