diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-arbiter b/for_fedora/init.d/shinken-arbiter
--- a/for_fedora/init.d/shinken-arbiter	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-arbiter	2013-02-19 11:41:38.095048977 +0100
@@ -43,7 +43,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-broker b/for_fedora/init.d/shinken-broker
--- a/for_fedora/init.d/shinken-broker	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-broker	2013-02-19 11:41:38.170048116 +0100
@@ -44,7 +44,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-poller b/for_fedora/init.d/shinken-poller
--- a/for_fedora/init.d/shinken-poller	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-poller	2013-02-19 11:41:38.202047749 +0100
@@ -48,7 +48,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-reactionner b/for_fedora/init.d/shinken-reactionner
--- a/for_fedora/init.d/shinken-reactionner	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-reactionner	2013-02-19 11:41:38.254047152 +0100
@@ -44,7 +44,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-receiver b/for_fedora/init.d/shinken-receiver
--- a/for_fedora/init.d/shinken-receiver	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-receiver	2013-02-19 11:41:38.281046842 +0100
@@ -43,7 +43,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-scheduler b/for_fedora/init.d/shinken-scheduler
--- a/for_fedora/init.d/shinken-scheduler	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-scheduler	2013-02-19 11:41:38.334046234 +0100
@@ -44,7 +44,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
diff -U 3 -H -d -r -N -- a/for_fedora/init.d/shinken-skonf b/for_fedora/init.d/shinken-skonf
--- a/for_fedora/init.d/shinken-skonf	2013-02-11 15:08:26.000000000 +0100
+++ b/for_fedora/init.d/shinken-skonf	2013-02-19 11:41:38.436045063 +0100
@@ -43,7 +43,7 @@
 
     directory=$(dirname $pidfile)                                                                                                                                    
     [ ! -d $directory ] && mkdir -p $directory
-    chown nagios:root $directory
+    chown @user@:@group@ $directory
 
     echo -n $"Starting $prog: "
     $exec $daemon_args
