--- bin/init.d/shinken.orig	2012-03-13 12:52:25.000000000 +0100
+++ bin/init.d/shinken	2012-06-22 12:41:48.651744923 +0200
@@ -184,26 +184,13 @@
 #set -xv
 
  
-echo_success() {
-   log_end_msg 0 $*
-}
-
-echo_failure() {
-    log_end_msg 1 $*
-}
-
-#log_end_msg
-
-
 # Load the VERBOSE setting and other rcS variables
 [ -f /etc/default/rcS ] && . /etc/default/rcS
 
 # Source function library.
 [ -f /etc/rc.d/init.d/functions ] && . /etc/rc.d/init.d/functions
 
-[ -f /lib/lsb/init-functions ] && . /lib/lsb/init-functions
-
-
+#[ -f /lib/lsb/init-functions ] && . /lib/lsb/init-functions
 
 ################################################
 
@@ -266,7 +253,7 @@
     ps -p "$pid" >/dev/null 2>&1
     rc=$?
     if [ $rc != 0 ]; then
-        log_failure_msg  "$mod NOT RUNNING (process $pid doesn't exist ?)"
+        echo "$mod NOT RUNNING (process $pid doesn't exist ?)"
         return 1
     fi
     echo "$mod RUNNING (pid $pid)"
@@ -280,7 +267,7 @@
     mod=$1
     modfilepath="$BIN/shinken-${mod}"
     [ -e "$modfilepath" ] || {
-        log_failure_msg "FAILED: did not found $mod file ($modfilepath) ; are you sure shinken-$mod is installed ?"
+        echo "FAILED: did not found $mod file ($modfilepath) ; are you sure shinken-$mod is installed ?"
         return 5
     }
     [ "$DEBUG" = 1 ] && DEBUGCMD="--debug "$(getdebugfile "$mod")
@@ -379,11 +366,11 @@
 ############################
 
 do_start_() {
-    echo  "Starting $1: "
+    echo -n "Starting $1: "
     status=$(do_status "$1")
     rc=$?
     if [ $rc -eq 0 ]; then
-        log_warning_msg "Already running"
+        echo "Already running."
         return
     fi
     if test "$1" = "arbiter"
@@ -401,16 +388,18 @@
     startoutput=$(do_start "$1")
     rc=$?
     if [ $rc -eq 0 ]; then
-        echo_success  
+        success
+        echo
     else
+        failure
+        echo
         echo "$startoutput" 
-        echo_failure
     fi
     return $rc
 }
 
 do_stop_() {
-    echo  "Stopping $1"
+    echo -n "Stopping $1"
     statusoutput=$(do_status "$1")
     rc=$?
     if [ $rc -ne 0 ]; then
@@ -421,17 +410,18 @@
         [ $rc -ne 0 ] && failuremsg="Couldn't stop $1 : $stopoutput"
     fi
     if [ $rc -ne 0 ]; then
-        log_failure_msg "$failuremsg"
-        echo_failure
+        failure
+        echo
+        echo "$failuremsg"
     else
-        echo_success
+        success
+        echo
     fi
     return $rc
 }
 
 do_restart_() {
     mod="$1"
-    echo "Restarting $mod"
     if [ "$mod" = "arbiter" ]; then
         do_check_ "$mod"
         checkrc=$?
@@ -439,48 +429,60 @@
            return 1
         fi
     fi 
+    echo -n "Restarting $mod"
     stopoutput=$(do_stop "$mod")
     startoutput=$(do_start "$mod")
     rc=$?
     if [ $rc -eq 0 ]; then
-        echo_success
+        success
+        echo
     else
-        log_failure_msg "$startoutput"
-        echo_failure
+        failure
+        echo
+        echo "$startoutput"
     fi
     return $rc
 }
 
 do_status_() {
     mod=$1
-    echo  "Checking status of $mod"
-    do_status "$1"
+    echo -n "Checking status of $mod: "
+    statusoutput=$(do_status "$1")
     rc=$?
     if [ $rc -eq 0 ]; then
-        echo_success
+        success
+        echo
     else
-        echo_failure
+        failure
+        echo
+        echo "$statusoutput"
     fi
 
 }
 
 do_check_() {
-    echo "Doing config check"
+    echo -n "Doing config check"
     output=$(do_check "$1" 2>&1)
     rc=$?
     if [ $rc -eq 0 ]; then
-        echo_success
+        success
+        echo
+
     else
         check_res_file="/tmp/shinken_checkconfig_result"
         echo "$output" > "$check_res_file"
         output=$(echo "$output" | tail -1)
-        log_warning_msg "full result is in ${check_res_file}"
-        log_failure_msg "ConfigCheck failed: $output"
-        echo_failure
+        failure
+        echo
+        echo "full result is in ${check_res_file}"
+        echo "ConfigCheck failed: $output"
     fi
     return $rc
 }
-do_checkconfig_() { do_check_ "$1" ; }
+
+do_checkconfig_() {
+    do_check_ "$1"
+}
 
 
 ############################
