diff -rup a/source/frontend/renderfrontend.cpp b/source/frontend/renderfrontend.cpp
--- a/source/frontend/renderfrontend.cpp	2013-11-16 09:52:20.000000000 +0100
+++ b/source/frontend/renderfrontend.cpp	2013-12-17 16:52:09.279624145 +0100
@@ -160,8 +160,8 @@ void RenderFrontendBase::ConnectToBacken
 	if(resultobj != NULL)
 		*resultobj = result;
 
-	if(console != NULL)
-		Message2Console::InitInfo(result, console.get());
+	/*if(console != NULL)
+		Message2Console::InitInfo(result, console.get());*/
 }
 
 void RenderFrontendBase::DisconnectFromBackend(POVMSAddress backendaddress)
