diff -rup a/source/backend/povray.cpp b/source/backend/povray.cpp
--- a/source/backend/povray.cpp	2013-11-16 09:52:20.000000000 +0100
+++ b/source/backend/povray.cpp	2013-12-17 16:48:46.520573975 +0100
@@ -688,6 +688,11 @@ int main(int argc, char **argv)
 	int i = 0;
 
 	printf("Welcome to POV-Ray 3.7 SMP!\n");
+	printf("Command line: ");
+	for(int i = 0 ; i < argc ; i++) {
+		printf("%s ", argv[i]);
+	}
+	printf("\n");
 	fflush(stdout);
 
 //	char *nargv[2];
diff -rup a/vfe/unix/unixconsole.cpp b/vfe/unix/unixconsole.cpp
--- a/vfe/unix/unixconsole.cpp	2013-11-16 09:52:20.000000000 +0100
+++ b/vfe/unix/unixconsole.cpp	2013-12-17 16:50:32.772500176 +0100
@@ -368,6 +368,11 @@ int main (int argc, char **argv)
 	int               argc_copy=argc; /* because it might also be updated */
 
 	/*fprintf(stderr, "%s: This is a RELEASE CANDIDATE version of POV-Ray. General distribution is discouraged.\n", PACKAGE);*/
+	fprintf(stderr, "%s", "Command line: ");
+	for(int i = 0 ; i < argc_copy ; i++) {
+		fprintf(stderr, "%s ", argv_copy[i]);
+	}
+	fprintf(stderr, "\n");
 
 	// block some signals for this thread as well as those created afterwards
 	sigemptyset(&sigset);
